pkg/yurtmanager/controller/yurtappoverrider/yurtappoverrider_controller.go

```diff
@@ -0,0 +1,223 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=apps.openyurt.io,resources=yurtappoverriders,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=controllerrevisions,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a YurtAppOverrider object and makes changes based on the state read
+// and what is in the YurtAppOverrider.Spec
+func (r *ReconcileYurtAppOverrider) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	// Note !!!!!!!!!!
+	// We strongly recommend use Format() to  encapsulation because Format() can print logs by module
+	// @kadisi
+	klog.Infof(Format("Reconcile YurtAppOverrider %s/%s", request.Namespace, request.Name))
+
+	// Fetch the YurtAppOverrider instance
+	instance := &appsv1alpha1.YurtAppOverrider{}
+	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileYurtAppOverrider) updatePools(yao *appsv1alpha1.YurtAppOverrider) error {
+	pools := make(map[string]struct{})
+	for _, entry := range yao.Entries {
+		for _, pool := range entry.Pools {
+			pools[pool] = struct{}{}
+		}
+	}
+	if _, ok := pools["*"]; ok {
+		switch yao.Subject.Kind {
+		case "YurtAppSet":
+			instance := &appsv1alpha1.YurtAppSet{}
+			if err := r.Get(context.TODO(), types.NamespacedName{Namespace: yao.Namespace, Name: yao.Subject.Name}, instance); err != nil {
+				return err
+			}
+			for _, pool := range instance.Spec.Topology.Pools {
+				pools[pool.Name] = struct{}{}
+			}
+		case "YurtAppDaemon":
+			instance := &appsv1alpha1.YurtAppDaemon{}
+			if err := r.Get(context.TODO(), types.NamespacedName{Namespace: yao.Namespace, Name: yao.Subject.Name}, instance); err != nil {
+				return err
+			}
+			for _, pool := range instance.Status.NodePools {
+				pools[pool] = struct{}{}
+			}
+		default:
+			return errors.NewBadRequest("unknown subject kind")
+		}
+	}
+	for pool := range pools {
+		deployments := v1.DeploymentList{}
+		listOptions := client.MatchingLabels{"apps.openyurt.io/pool-name": pool}
+		err := r.List(context.TODO(), &deployments, listOptions)
+		if err != nil {
+			return err
+		}
+		for _, deployment := range deployments.Items {
+			if err := r.Client.Update(context.TODO(), &deployment); err != nil {
```

Comment:
I think it's a good idea to add an annotation for deployment for specifying the update timestamp of yurtappoverrider.